llvm/include/llvm/CodeGen/SelectionDAGNodes.h

```diff
@@ -3010,6 +3010,22 @@ class MaskedHistogramSDNode : public MaskedGatherScatterSDNode {
   }
 };
 
+class PartialReduceAddSDNode : public SDNode {
+public:
+  friend class SelectionDAG;
```

Comment:
A special class is probably not needed (most nodes don't have these), and I think there may be some issues (see next comment).

Reply:
Done